og_access/src/BundleFormAlter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\og_access;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\og\Og;
+
+/**
+ * Helper for og_access_form_alter().
+ */
+class BundleFormAlter {
+
+  /**
+   * The entity bundle.
+   *
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * The entity type ID.
+   *
+   * @var string
+   */
+  protected $entityTypeId;
+
+  /**
+   * The form entity which has been used for populating form element defaults.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $entity;
+
+  /**
+   * Construct a BundleFormAlter object.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity object.
+   */
+  public function __construct(EntityInterface $entity) {
+    $this->entity = $entity;
+  }
+
+  /**
+   * This is a helper for og_ui_form_alter().
+   *
+   * @param array $form
+   *   The form variable.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function formAlter(array &$form, FormStateInterface $form_state) {
+    // Example: node.
+    $this->entityTypeId = $this->entity->getEntityType()->getBundleOf();
+
+    // Example: article.
+    $this->bundle = $this->entity->id();
+
+    $form['og']['og_enable_access'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Enable OG access control'),
```

Comment:
'Enable' is not a very good word for checkboxes. Something like 'Restrict access to group members' would be clearer.

It also needs description text to say more precisely what it does. I assumed this would restrict access to the group AND its content, but as I also see this checkbox on my group content bundle form, I assume it's actually just restricting access to the bundle it's enabled on.

Reply:
> I assumed this would restrict access to the group AND its content
@zerolab, is this assumption correct?

Looking at the following code, I think it's not implemented correct:
```
  if ($node->hasField(OgAccess::OG_ACCESS_CONTENT_FIELD) &&
      !empty($node->get(OgAccess::OG_ACCESS_CONTENT_FIELD))) {
    $content_access = $node->get(OgAccess::OG_ACCESS_CONTENT_FIELD)->value;
  }
  else {
    $content_access = OgAccess::OG_ACCESS_PUBLIC;
  }
```

Setting a group to private should mark all group content as private (except when the group content bundle has an OG_ACCESS_CONTENT_FIELD there's might be an override).

Reply:
Note that this is a direct port of the D7 code. The same logic is in D7, if I recall this correctly